File: Assets/Scripts/MoveAround.cs
using UnityEngine;
using System.Collections;

[RequireComponent (typeof (CharacterController))]


public class MoveAround : MonoBehaviour {

	public float 		speed = 3.0f;
	public float 		rotateSpeed = 3.0f;

	private Vector3 	forward;
	private float 		curSpeed;

	// Update is called once per frame
	void Update () {

		CharacterController controller = GetComponent <CharacterController>();

		// rotate around the y-axis
		transform.Rotate (0, Input.GetAxis ("Horizontal") * rotateSpeed, 0);

		// move forward/backward

		forward = transform.TransformDirection (Vector3.forward);
		curSpeed = speed * Input.GetAxis ("Vertical");
		controller.SimpleMove (forward * curSpeed);

	}

}

File: Assets/Scripts/MoveAround.js
﻿#pragma strict

var speed : float = 3.0;
var rotateSpeed : float = 3.0;

var moveJoystick : Joystick;
var rotateJoystick : Joystick;

var originalPosition : Vector3;
var originalRotation : Quaternion;


function Awake() {
	originalPosition = transform.position;
	originalRotation = transform.rotation;

}



function Update () {

	
	if (GameController != null && !GameController.gameRunning)
		return;

	var controller : CharacterController = GetComponent(CharacterController);
	
	//rotate around y-axis
	var rotatePos = Input.GetAxis ("Horizontal") ? Input.GetAxis ("Horizontal") : joyStickInput(rotateJoystick);
	transform.Rotate(0, rotatePos * rotateSpeed, 0);
	
	//move forward/backward
	var forward = transform.TransformDirection(Vector3.forward);
	var movePos = Input.GetAxis ("Vertical") ? Input.GetAxis ("Vertical") : joyStickInput(moveJoystick);
	
	var curSpeed : float = speed * movePos;
	
	controller.SimpleMove(forward * curSpeed);
}



function joyStickInput (joystick : Joystick) {
	var absJoyPos = Vector2 (Mathf.Abs(joystick.position.x), Mathf.Abs(joystick.position.y));
	
	var xDirection = (joystick.position.x > 0) ? 1:-1;
	var yDirection = (joystick.position.y > 0) ? 1:-1;
	
	return ( ( absJoyPos.x > absJoyPos.y) ? absJoyPos.x * xDirection : absJoyPos.y * yDirection);
}

function resetGame() {
	// reset to original position
	transform.position = originalPosition;
	transform.rotation = originalRotation;

}

@script RequireComponent(CharacterController)